app/models/user.rb

```diff
@@ -90,7 +90,7 @@ class User < ApplicationRecord
   has_many :identities_enabled, -> { enabled }, class_name: "Identity", inverse_of: false
   has_many :mentions, dependent: :destroy
   has_many :messages, dependent: :destroy
-  has_many :notes, as: :noteable, inverse_of: :noteable
+  has_many :notes, as: :noteable, inverse_of: :noteable, dependent: :nullify
```

Comment:
Apparently we keep notes around, not sure why but: 

https://github.com/forem/forem/blob/2fe4a36271106f22a19ef996f5432e58d249b2b9/spec/services/users/delete_spec.rb#L91-L99